drivers/digital-io/max149x6/iio_max14916.c

```diff
@@ -0,0 +1,296 @@
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max14916_iio_read_scale(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+	int32_t val = 1;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+}
+
+int max14916_iio_setup_channels(struct max14916_iio_desc *desc)
+{
+	struct iio_channel *max14916_iio_channels;
+	uint32_t enabled_ch = 0;
+	uint32_t ch_offset = 0;
+	uint32_t i;
+
+	for (i = 0; i < MAX14916_CHANNELS; i++)
+		if (desc->ch_enable[i])
+			enabled_ch++;
+
+	max14916_iio_channels = no_os_calloc(enabled_ch,
+					     sizeof(*max14916_iio_channels));
+	if (!max14916_iio_channels)
+		return -ENOMEM;
+
+	for (i = 0; i < MAX14916_CHANNELS; i++) {
+		if (desc->ch_enable[i]) {
+			max14916_iio_channels[ch_offset] = (struct iio_channel)MAX14916_CHANNEL(i);
+			max14916_iio_channels[ch_offset].attributes = max14916_attrs;
+			ch_offset++;
+		}
+		max14916_iio_channels[i].ch_out = 1;
+	}
+
+	desc->iio_dev->channels = max14916_iio_channels;
+	desc->iio_dev->num_ch = enabled_ch;
+
+	return 0;
+}
+
+/**
+ * @brief Register read wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ * @param readval - Register value
+ * @return 0 in case of success, error code otherwise
+ */
+static int max14916_iio_reg_read(struct max14916_iio_desc *dev, uint32_t reg,
+				 uint32_t *readval)
+{
+	return max149x6_reg_read(dev->max14916_desc, reg, readval);
+}
+
+/**
+ * @brief Register write wrapper
+ * @param dev - The iio device structure.
+ * @param reg - The register's address.
+ * @param readval - Register value
+ * @return 0 in case of success, error code otherwise
+ */
+static int max14916_iio_reg_write(struct max14916_iio_desc *dev, uint32_t reg,
+				  uint32_t writeval)
+{
+	return max149x6_reg_write(dev->max14916_desc, reg, writeval);
+}
+
+/**
+ * @brief Initializes the MAX14916 IIO descriptor.
+ * @param iio_desc - The iio device descriptor.
+ * @param init_param - The structure that contains the device initial parameters.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int max14916_iio_init(struct max14916_iio_desc **iio_desc,
+		      struct max14916_iio_desc_init_param *init_param)
+{
+	struct max14916_iio_desc *descriptor;
+	int ret;
+
+	if (!init_param || !init_param->max14916_init_param)
+		return -EINVAL;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = max14916_init(&descriptor->max14916_desc,
+			    init_param->max14916_init_param);
+	if (ret)
+		goto free_desc;
+
+	descriptor->iio_dev = &max14916_iio_dev;
+
+	memcpy(descriptor->ch_enable, init_param->ch_enable,
```

Comment:
You don't need to store `ch_enable` in the descriptor, as it's only used during init

Reply:
ch_enable will be stored only in the init_param since it is not needed only during init.